es/index.js

```diff
@@ -16,4 +16,10 @@ if (typeof self !== 'undefined') {
 }
 
 var result = ponyfill(root);
+
+root.Symbol = root.Symbol || function Symbol(description) {
+  return '@@' + description;
```

Comment:
The only problem I see with this is that it doesn't align with what `Symbol` does.

With `Symbol(description)` the description doesn't affect the equality of the result. In other words: `Symbol('test') !== Symbol('test')`... This is different than `Symbol.for`, which does what you're expecting this to do.

The result of `Symbol` must be a completely unique value. The `@@` helps, and is an accepted practice. But really, I guess you'd need to do something like:

```ts
let _symbolNumber = 0;
function Symbol(description) {
  return `@@_${_symbolNumber++}_${description}`
}
```

The other depressing bit about this, is you'd have to make sure that `description` was converted into a valid set of characters for a property name. OR you'd have to just attach the description to the string you returned somehow (although I haven't thought much about how).

Since we're polyfilling `Symbol` in this case, I guess I'd expect it to behave appropriately (and even possibly implement `Symbol.for`)

Reply:
Ah, and the code I’m modifying presumes the existence of `Symbol` or a `Symbol` shim as you describe. That being the case, I think it would be sufficient for parity to simply remove this line. Does that track?

Reply:
We could remove that line yeah... However, I'm not sure if there's a runtime that exists where `WeakMap` exists, but `Symbol` does not. I'm trying to research that now.

So I have two questions:

1. Can you elaborate on the problem you're trying to solve?
2. Can we move the creation of the `WeakMap` inside of the check for existence of `Symbol`?

Reply:
I guess, if I'm understanding right, in a runtime you're using calling `Symbol.observable = ???` will throw an error because `Symbol` is frozen. Is that correct?

If that's the case, could we just wrap it in a try-catch?

Reply:
We could remove the `WeakMap` but I would have to make a more invasive change to the test, either by exporting a `clearForRetest` function or running each test in a fresh process.

Capturing the failed assignment to `Symbol.observable` is an option. Checking the property descriptor for writability would be another. This is still my recommendation since it clearly separates the polyfill from the ponyfill. I’ve really just moved the assignment to `Symbol.observable` from the ponyfill to the polyfill. I’ll write up whatever solution you prefer.

Reply:
>  if there's a runtime that exists where WeakMap exists, but Symbol does not. I'm trying to research that now.

They were both introduced in ES6, so I would be surprised. Please let us know what you find. I'm curious.

Reply:
> I guess, if I'm understanding right, in a runtime you're using calling `Symbol.observable = ???` will throw an error because `Symbol` is frozen. Is that correct?

Yes.

> If that's the case, could we just wrap it in a try-catch?

And do what in the catch clause? Whatever that is, if it works in that case, why not just do that all the time?